ctapipe/io/dl1writer.py

```diff
@@ -468,6 +487,34 @@ def fill_from_simtel(
                         containers=hist_container,
                     )
 
+    def _write_muon_events(self, writer: TableWriter, event: ArrayEventContainer):
+        """
+        add muon entries to the event/telescope tables for each muon event
+        """
+
+        # write the telescope tables
+
+        for tel_id, dl1_camera in event.dl1.tel.items():
+            dl1_camera.prefix = ""  # don't want a prefix for this container
+            telescope = self._subarray.tel[tel_id]
+            self.log.debug("WRITING TELESCOPE %s: %s", tel_id, telescope)
+
+            tel_index = TelEventIndexContainer(
+                obs_id=event.index.obs_id,
+                event_id=event.index.event_id,
+                tel_id=np.int16(tel_id),
+            )
+
+            if dl1_camera.muon_parameters is not None:
+                writer.write(
+                    table_name=f"dl1/event/telescope/parameters/muons",
+                    containers=[tel_index, *dl1_camera.muon_parameters.values()],
+                )
+
+        writer.write(
```

Comment:
If this is here, is it written twice if we save both muon parameters and images / normal image parameters?

Also it looks like an "if is_simulation" is missing

Reply:
In the current form, a is_muon-flag is set on writer creation, so one instance can only write muons or regular events and this problem should not occure (General Question: Is this wanted behaviour or should one writer instance be able to write both types?). However, since this information is written for both types, having these lines in `__call__` makes things easier.

Reply:
> General Question: Is this wanted behaviour or should one writer instance be able to write both types 

Yes, we should not have two writer instances writing to the same file. I think this does not even work, since hdf actually blocks files for writing.


Just adding a `write_muon_parameters(self, event)` method seems best to me.

Reply:
`simulation/event/subarray/shower` is where the simulation parameters should go, but you shouldn't need to touch that in the muon component - the DL1Writer will put it in the correct place. 

Basically, the old ctapipe-reconstruct-muons script did not follow any consistant data model, so there is no need to have a new tool reproduce the same format.